### PR TITLE
libcgroup: 0.42.2 -> 2.0.2

### PR DESCRIPTION
**Pull request by karantan**
###### Description of changes

Update libcgroup package to v2.0.2.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NorfairKing** [comment]
I think this diff fixes the build:

```
diff --git a/pkgs/os-specific/linux/libcgroup/default.nix b/pkgs/os-specific/linux/libcgroup/default.nix
index fb72be4f69f..715aa690277 100644
--- a/pkgs/os-specific/linux/libcgroup/default.nix
+++ b/pkgs/os-specific/linux/libcgroup/default.nix
@@ -8,7 +8,8 @@ stdenv.mkDerivation rec {
     owner = pname;
     repo = pname;
     rev = "v${version}";
-    sha256 = "0bw2xsmkpaj1qgv0kgmyp4yhfrknh4l497i00ndk74bjajj09pjr";
+    hash = "sha256-o9eXbsgtGhODEbpbEn30RbYfYpXo6xkU5ptU1och5tU=";
+    fetchSubmodules = true;
   };
 
   buildInputs = [ pam bison flex ];
```

**Comment by karantan** [comment]
> I think this diff fixes the build:
> 
> ```
> diff --git a/pkgs/os-specific/linux/libcgroup/default.nix b/pkgs/os-specific/linux/libcgroup/default.nix
> index fb72be4f69f..715aa690277 100644
> --- a/pkgs/os-specific/linux/libcgroup/default.nix
> +++ b/pkgs/os-specific/linux/libcgroup/default.nix
> @@ -8,7 +8,8 @@ stdenv.mkDerivation rec {
>      owner = pname;
>      repo = pname;
>      rev = "v${version}";
> -    sha256 = "0bw2xsmkpaj1qgv0kgmyp4yhfrknh4l497i00ndk74bjajj09pjr";
> +    hash = "sha256-o9eXbsgtGhODEbpbEn30RbYfYpXo6xkU5ptU1och5tU=";
> +    fetchSubmodules = true;
>    };
>  
>    buildInputs = [ pam bison flex ];
> ```

thanks, it worked

File: pkgs/os-specific/linux/libcgroup/default.nix [hunk 1]
```diff
@@ -2,13 +2,14 @@
...
-  version = "0.42.2";
+  version = "2.0.2";
 
   src = fetchFromGitHub {
     owner = pname;
     repo = pname;
     rev = "v${version}";
-    sha256 = "1h8s70lm6g7r0wj7j3xgj2g3j9fifvsy2pna6w0j3i5hh42qfms4";
+    hash = "sha256-o9eXbsgtGhODEbpbEn30RbYfYpXo6xkU5ptU1och5tU=";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/libcgroup/default.nix:12]:
```suggestion
    fetchSubmodules = true;
    hash = "sha256-o9eXbsgtGhODEbpbEn30RbYfYpXo6xkU5ptU1och5tU=";
```

**Comment by SuperSandro2000** [comment]
@ofborg build libcgroup

**Pull request merged**